test/config.js

```diff
@@ -0,0 +1,85 @@
+import path from 'path';
+import os from 'os';
+import test from 'ava';
+import sinon from 'sinon';
+import proxyquire from 'proxyquire';
+
+const homedirStub = sinon.stub(os, 'homedir');
+const fixtureBasePath = path.resolve('test', 'fixtures', 'config');
+
+test('should return config from `.np-config.json` in home-directory when global binary used', async t => {
+	homedirStub.returns(path.resolve(fixtureBasePath, 'homedir1'));
+	const config = proxyquire('../source/config', {
+		'is-installed-globally': true,
+		'pkg-dir': async () => {
+			return path.resolve(fixtureBasePath, 'pkg-dir');
+		}
+	});
+	t.deepEqual(await config(), {yarn: false});
```

Comment:
This only verifies that we use the default value of `yarn` instead of the one specified in the fixture `package.json` config.
This test should also verify that we _do_ use the value specified in the global config.
Also, I'd prefer it if the tests used a string-based option such as `tag`, to avoid situations such as this one and to increase their effectiveness.

Reply:
(Or just define `local`/`global`/anything else for the purpose of testing, as you did in some of the tests below.)

Reply:
I added a unique `source`-property for every fixture-file.